test/hotspot/jtreg/runtime/jfr/TestJFRDiskFull.java

```diff
@@ -0,0 +1,127 @@
+    @Description("JFRDiskFull Event")
+    static class JFRDiskFullEvent extends Event {
+        @Label("Message")
+        String message;
+    }
+
+    private static final long LEFT_SIZE = 307200L;
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 1) {
+            Configuration profConfig = Configuration.getConfiguration("profile");
+            Recording recording = new Recording(profConfig);
+            recording.setName("TestJFR");
+            recording.start();
+            for (int i = 0; i < 1000000000; i++) {
+                try {
+                    JFRDiskFullEvent event = new JFRDiskFullEvent();
+                    event.message = "JFRDiskFull";
+                    event.begin();
+                    event.commit();
+                } catch (Exception ex) {
+                }
+            }
+            recording.stop();
+            System.out.println("should not reach here");
+        } else {
+            runtest();
+        }
+    }
+
+    private static void runtest() throws Exception {
+        // disk full
+        File file = File.createTempFile("largefile", null, new File("."));
+        file.deleteOnExit();
+        long spaceavailable = file.getUsableSpace();
+        System.out.println("spaceavailable = " + spaceavailable);
+        long filesize = spaceavailable - LEFT_SIZE;
+        if (filesize > 0) {
+            createLargeFile(filesize, file);
+        }
+        System.out.println("spaceavailable = " + file.getUsableSpace());
+
+        ProcessBuilder pb = ProcessTools.createJavaProcessBuilder("-XX:FlightRecorderOptions=maxchunksize=1M",
+                "TestJFRDiskFull", "Recording");
+        OutputAnalyzer oa = ProcessTools.executeProcess(pb);
+        long pid = oa.pid();
+        oa.shouldMatch(
+                "\\[[0-9]+\\.[0-9]+s\\]\\[error\\]\\[jfr,system\\] Failed to write to jfr stream because no space left on device")
+                .shouldMatch(
+                        "\\[[0-9]+\\.[0-9]+s\\]\\[error\\]\\[jfr,system\\] An irrecoverable error in Jfr. Shutting down VM\\.\\.\\.")
+                .shouldNotContain("Internal Error")
+                .shouldNotContain("should not reach here");
+
+        File errLog = new File("hs_err_pid" + pid + "\\.log");
+        if (errLog.exists()) {
+            throw new RuntimeException("hs_err log is created");
+        }
+    }
+
+    private static void createLargeFile(long filesize, File file) throws Exception {
+        file.delete();
+        RandomAccessFile raf = new RandomAccessFile(file, "rw");
+        byte[] x = new byte[1024];
+        System.out.println("  Writing large file...");
+        long t0 = System.nanoTime();
+        while (file.length() < filesize) {
+            raf.write(x);
+        }
+        long t1 = System.nanoTime();
+        System.out.printf("  Wrote large file in %d ns (%d ms) %n", t1 - t0, TimeUnit.NANOSECONDS.toMillis(t1 - t0));
+        raf.close();
+    }
+}
```

Comment:
I appreciate the effort, but we can't have a test that intentionally provokes a disk full situation. The updated error message will have to be manually inspected instead.

Reply:
I use `@run main/manual` in TestJFRDiskFull.java. I think this label means manually test.  
I mannually confirmed this test to pass with jtreg after this fix.

Reply:
My apologies, I missed the @run main/manual decoration. I don't think we have any JFR tests that use it.

If you can ensure this test is excluded for automatic runs, then perhaps...but then I don't know who will get to run it, so the value of the test is iquestionable.

Reply:
Manual tests are excluded if the jtreg test run specifies to run automatic tests only (as we do in our CI). So this really only serves as a validation of the fix, with no real expectation that anyone will necessarily every run it again. Even as a locally run test, filling the disk can easily lead to unexpected problems for other processes - including the swap/paging file on Windows - so this is a risky test to run.